crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -241,7 +241,7 @@ pub(crate) enum CallBindingError<'db> {
     MissingArguments { parameters: ParameterContexts },
     /// A call argument can't be matched to any parameter.
     UnknownArgument {
-        argument_name: ast::name::Name,
+        argument_name: &'a str,
```

Comment:
I'd use a `Name` here as constructing an error should be the exception. That allows removing the lifetime from `CallBindingError`, `CallBinding`, `CallDunderResult`, significantly reducing the size of the diff.

Reply:
Nice, thanks! It's much simpler now